### PR TITLE
SI-6039 Harden against irrelevant filesystem details

### PR DESCRIPTION
**Pull request by paulp**
The symbol loader need not create and populate package symbols merely because
there is a directory somewhere. Every package created based on the existence
of a directory should contain a classfile, either directly or indirectly.

### CODE REVIEW

File: test/files/run/t6039.scala [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    new java.io.File("util") mkdirs
+    val classpath = List(sys.props("partest.lib"), ".") mkString sys.props("path.separator")
+    log(s"classpath = $classpath")
+    compileString(newCompiler("-cp", classpath, "-d", testOutput.path))(packageCode)
+  }
+  def code = ???
+  def packageCode = """
+package scala.bippy
+class A { new util.Random() }
+"""
```
Comment (phaller) [inline test/files/run/t6039.scala:14]: Why not use `stripMargin`?

```
def packageCode = """
  |package scala.bippy
  |class A { new util.Random() }
  |""".stripMargin('|')
```

yields the same result.

Comment (paulp) [inline test/files/run/t6039.scala:14]: ...right up until you want to paste the source from the file into something else. In a test case it doesn't seem particularly pressing that the indentation be perfect.

Comment (phaller) [inline test/files/run/t6039.scala:14]: Good point. I just couldn't resist. ;-)

Comment (paulp) [inline test/files/run/t6039.scala:14]: Hey, I'll take what review I can get around here. Speaking of which, can I somehow interpret that as a LGTM?

File: src/compiler/scala/tools/nsc/symtab/SymbolLoaders.scala [hunk 2]
```diff
@@ -234,10 +234,14 @@ abstract class SymbolLoaders {
         }
       }
       if (!root.isEmptyPackageClass) {
+        // Only enter packages which contain a class or a non-empty package
         for (pkg <- classpath.packages) {
-          enterPackage(root, pkg.name, new PackageLoader(pkg))
+          if (pkg.isEmptyOfClassfiles) {
+            log(s"Discarding $root/$pkg as it contains no classfiles.")
```
Comment (xeno-by) [inline src/compiler/scala/tools/nsc/symtab/SymbolLoaders.scala:240]: This is not just useful for hacking scalac, but is also consistent with Java's behavior. As SLS 4.7 hints, 

```
The import clause import p._ is equivalent to import p.{_}, i.e. it makes available without qualification all members of p (this is analogous to import p.* in Java).
```

LGTM

**Comment by retronym** [comment]
LGTM.

**Comment by som-snytt** [comment]
Reversion therapy https://github.com/scala/scala/pull/2614

**Pull request merged**